### DISCUSSION TITLE
Support for EON Landis+Gyr Modell: E360

### DISCUSSION THREAD

**Discussion by mfrejman** [UtilitechAS/amsreader-firmware#447] [Q&A]
Hi I have a 
Landis+Gyr 
Modell: E360
And try to get it to work, but no luck :-(
My meter is not recognized in the web gui configuration
any tips how to solve the issue 

I'm using ESP8266 Wemos D1 mini
HW connection setup with power from the meter 
from: https://github.com/UdoK/esp8266_p1meter_sv

link to EON meter
https://www.eon.se/content/dam/eon-se/swe-documents/swe-manual-trefasmatare-f3-100_1.pdf

Best regards

**Comment by ArnieO** [comment]
1. There is an FTDI chip on the Wemos that uses the Rx input, so I don't think you can use that input. And I don't think you'll be able to use one of the other GPIOs at the required serial speed 115 200 baud.
You could try a Wemos with ESP32S2, and use an ordinary GPIO. On that chip, serial can be swapped to any GPIO.
2. On some meters you need to reduce the pullup resistor to around 3k when you pull to 3,3V to get a reliable signal.

**Reply by gskjold** [reply]
Correct, connecting to RX will not work on a dev board, but you can be connected to GPIO13 (D7) and choose UART2 as HAN pin. Note that this will effectively disable debug output via USB, as the UART pins are switched to 13 and 15.

**Reply by mfrejman** [reply]
I changed to D7 and set the data request to +5 
https://hanporten.se/svenska/porten/
changed to UART2 in the setup 
for the meter
115200 8N1 

but no data and HAN icon is red
![image](https://user-images.githubusercontent.com/80361543/218563209-cc3b47b9-d0db-43dd-94fe-7422b469d01d.png)

**Reply by ArnieO** [reply]
The low voltage is not a good sign.
Did you change the pullup resistor to a lower value?

**Reply by mfrejman** [reply]
I change to the closes resistor I had 4,7k
but no change.
What is the ESP voltage measure (internal sensor) or input signal? 

I also found another wiring diagram 
https://github.com/psvanstrom/esphome-p1reader
and it is using the 5v as pullup.

Any suggestions?

Also thanks for all the help and input :-)

**Reply by gskjold** [reply]
Any chance you could enable telnet debugging and check if it receives anything? (https://amsleser.no/blog/post/24-telnet-debug) 

It should show an error in the GUI for invalid data if it does actually receive something, but it would be good to check with verbose level debugging to verify.

**Reply by ArnieO** [reply]
> I change to the closes resistor I had 4,7k but no change. What is the ESP voltage measure (internal sensor) or input signal?
> 
> I also found another wiring diagram https://github.com/psvanstrom/esphome-p1reader and it is using the 5v as pullup.
> 
> Any suggestions?
> 
> Also thanks for all the help and input :-)

When the signal is sent to a 3,3V microcontroller you shold pullup to 3,3V - so your implementation is correct. 
On ESP8266 the ESP voltage is an internal measurement in the ESP module. 

"Stupid" question, just in case: Have you contacted your grid company to have the HAN port activated? (It is by default disabled.)

**Reply by mfrejman** [reply]
HAN port is enabled, for EON you can enable the port on the web or in the App.

**Reply by mfrejman** [reply]
Just connected to usb and upgraded to new firmware
ESP low even connected to usb power 
![image](https://user-images.githubusercontent.com/80361543/218810207-e476e272-34c8-4319-b706-6057d278409d.png)

I will try and flash another wmos D1 mini and see if that solve the issue.
take care

**Reply by ArnieO** [reply]
Yes, the low voltage is an indication something is wrong. Does it correspond to voltage measured on the ESP? 
You might need to reconfigure after the update. Add "/vendor" to the URL. Select "Generic ESP8266" and the right GPIO for HAN - then press Save. This will give you a "Hardware" section on the config page that shows GPIO settings etc, like the ones you had on System/GPIO on firmware v2.1.x:
![image](https://user-images.githubusercontent.com/10295178/218812810-c36e62a8-5b3f-4baf-92e5-0855cfc0ecaf.png)

If measured voltage is OK, it could be that the ADC is badly calibrated on your device - in which case you should adjust the "Multiplier" field in Config/Hardware. (Espressif ESP32 modules are easier, they come flashed with calibration data - so no user calibration of ADC is needed.)

And please run a Telnet debug as suggested by @gskjold : https://amsleser.no/blog/post/24-telnet-debug)

**Reply by gskjold** [reply]
D1 mini need 1.1 on multiplier. More on that here: https://github.com/UtilitechAS/amsreader-firmware/wiki/FAQ

**Reply by mfrejman** [reply]
changed to D1 mini and now the ESP is green 
![image](https://user-images.githubusercontent.com/80361543/218817895-c28c2734-fd42-4275-a22d-7efc28be9d24.png)
